src/aui/auibook.cpp

```diff
@@ -2154,6 +2154,22 @@ bool wxAuiNotebook::RemovePage(size_t page_idx)
     // set new active pane unless we're being destroyed anyhow
     if (new_active && !m_isBeingDeleted)
         SetSelectionToWindow(new_active);
+        
+    // If last page was removed generate selection change events
+    // that usually produce in SetSelection method. See ticket #9920.
+    if (m_tabs.GetPageCount() == 0)
+    {
+        wxAuiNotebookEvent evt(wxEVT_AUINOTEBOOK_PAGE_CHANGING, m_windowId);
+        evt.SetSelection(-1);
+        evt.SetOldSelection(page_idx);
+        evt.SetEventObject(this);
+        GetEventHandler()->ProcessEvent(evt);
+        if (evt.IsAllowed())
```

Comment:
Hmm, this event can't be really disallowed, of course, so I'm not sure if we really want to test for this.

Could you please check how does it work for the native `wxNotebook`? I.e. what happens if we veto the `CHANGING` event for it, do we still get the `CHANGED` one?

TIA!

Reply:
I checked native control and it don't generate page change* events after removing last page, in other case at least on MSW it [send](https://github.com/wxWidgets/wxWidgets/blob/master/src/msw/notebook.cpp#L327-L337) changed event only if changing event wasn't vetoed. 

I checked native control and it don't generate page change* events after removing last page. 
But, if remove any page and calling Veto in PAGE_CHANGING handler native control won't send this event anymore even when I delete rest pages (tested on Win10). Very strange.

Reply:
I've tested notebook sample under GTK and it doesn't send the event when deleting the last page neither. So now I'm really not sure what to do: should we change the behaviour under all platforms to send a possibly unexpected (because it doesn't have valid selection) event or avoid changing `wxAuiNotebook` too to keep it consistent with the native controls?

Reply:
I think it will be better keep consistent behaviour and update the documentation. But I'm not an expert :)